### PR TITLE
[NO-ISSUE] Added some additional unit tests for queue utils

### PR DESCRIPTION
**Pull request by charlieduong94**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,6 +38,7 @@
     "fashion-model-defaults": "^1.0.1",
     "js-yaml": "^3.8.4",
     "msgpack-lite": "^0.1.26",
+    "proxyquire": "^1.8.0",
```
Comment (Druotic) [inline package.json]: This should be a dev dependency

Comment (austinkelleher) [inline package.json]: This should be a `devDependency`

File: test/unit/queue/util/createConsumer-test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      queueName: 'test-queue'
+    }
+  })
+
+  // simulate connection failure
+  connection.emit('close')
+  await Promise.delay(100)
+
+  // a new connection
+  t.true(spy.calledTwice)
```
Comment (Druotic) [inline test/unit/queue/util/createConsumer-test.js]: Not a huge deal, but might be useful to check the args too.

File: test/unit/queue/util/createProducer-test.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+test('should throw an error if now queueName is provided', async (t) => {
+  try {
+    await createProducer({
+      logger: console,
+      amqUrl: TEST_AMQ_URL
+    })
+    t.fail()
+  } catch (err) {
+    console.log('!!!!!!!!!!!!!!!!!!!!!', err)
```
Comment (austinkelleher) [inline test/unit/queue/util/createProducer-test.js]: Remove

**Comment by austinkelleher** [comment]
Besides moving `proxyquire` to be a devDev and removing the `console.log`, this looks good to me. Thanks for writing these additional tests!

File: test/unit/queue/util/createProducer-test.js (hunk 3 above)
Comment (charlieduong94) [inline test/unit/queue/util/createProducer-test.js]: Whoops thanks for pointing this out

File: package.json (hunk 1 above)
Comment (charlieduong94) [inline package.json]: Nice catch, thanks

**Pull request merged**